src/server/_common.py

```diff
@@ -18,12 +20,37 @@ def _get_db() -> Connection:
         g.db = conn
     return g.db
 
+def log_and_raise_exception(message):
+    app.logger.error(message)
+    raise ValidationFailedException(message)
 
 """
 access to the SQL Alchemy connection for this request
 """
 db: Connection = cast(Connection, LocalProxy(_get_db))
 
+@event.listens_for(engine, "before_cursor_execute")
+def before_cursor_execute(conn, cursor, statement, parameters, context, executemany):
+    t = time.time()
+    # TODO: probbly just use one of these two:
+    context._query_start_time = t
+    conn.info.setdefault('query_start_time', []).append(t)
```

Comment:
do we have pros/cons for each?

Reply:
The timing for both of these is identical and they use a similar global variables mechanism. I'll leave in `context._query_start_time` for consistency with API request timing.

Reply:
theyre both effectively doing the same thing, but i dont recall semantic differences or what other implications might be.  the naming of the member in `context` (with the leading underscore) implies to me that it is more non-standard and [this documentation](https://docs.sqlalchemy.org/en/14/core/events.html#sqlalchemy.events.ConnectionEvents.before_cursor_execute) indicates `context` could sometimes even be `None`.  on the other hand, the `info` member of `conn` is used here in a way that suggests that it is designed for such things, but the array also might mean we will have multiple accessors?  should the array be a thread-safe dict, keyed on pid?

things will probably work out fine if we go with either choice, but maybe some experimentation or research would be prudent.